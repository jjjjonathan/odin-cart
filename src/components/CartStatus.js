const CartStatus = (props) => {
  if (props.quantityInCart === 0) {
    return (
      <div className="item-form">
        <form noValidate onSubmit={props.onAddToCart} data-index={props.index}>
          <label htmlFor="quantity">Qty:</label>
          <select name="quantity" id="quantity">
            <option value="1">1</option>
            <option value="2">2</option>
            <option value="3">3</option>
            <option value="4">4</option>
            <option value="5">5</option>
            <option value="6">6</option>
          </select>
          <button type="submit">Add to cart</button>
        </form>
      </div>
    );
  } else {
    return (
      <div className="item-form">
        <p className="success">Added {props.quantityInCart} to cart!</p>
        <button
          type="button"
          onClick={props.onRemoveFromCart}
          data-index={props.index}
        >
          Remove from cart
        </button>
      </div>
    );
  }
};

export default CartStatus;
