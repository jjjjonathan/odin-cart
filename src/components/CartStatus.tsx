import React, { useState, useEffect } from 'react';

export type AddToCartHandler = (index: number, newQuantity: number) => void;
export type RemoveFromCartHandler = (index: number) => void;

type CartStatusProps = {
  quantityInCart: number;
  onAddToCart: AddToCartHandler;
  onRemoveFromCart: RemoveFromCartHandler;
  index: number;
};

const CartStatus = ({
  quantityInCart,
  onAddToCart,
  onRemoveFromCart,
  index,
}: CartStatusProps) => {
  const [selectedQuantity, setSelectedQuantity] = useState<string>('1');

  useEffect(() => {
    if (quantityInCart > 0) setSelectedQuantity(quantityInCart.toString());
  }, [quantityInCart]);

  const handleAddToCart = (event: React.MouseEvent<HTMLButtonElement>) => {
    event.preventDefault();
    const newQuantity = parseInt(selectedQuantity);
    onAddToCart(index, newQuantity);
  };

  if (quantityInCart === 0) {
    return (
      <div className="item-form">
        <label htmlFor="quantity">Qty:</label>
        <select
          value={selectedQuantity}
          onChange={(e) => setSelectedQuantity(e.target.value)}
          name="quantity"
          id="quantity"
        >
          <option value="1">1</option>
          <option value="2">2</option>
          <option value="3">3</option>
          <option value="4">4</option>
          <option value="5">5</option>
          <option value="6">6</option>
        </select>
        <button onClick={handleAddToCart}>Add to cart</button>
      </div>
    );
  } else {
    return (
      <div className="item-form">
        <p className="success">Added {quantityInCart} to cart!</p>
        <button type="button" onClick={() => onRemoveFromCart(index)}>
          Remove from cart
        </button>
      </div>
    );
  }
};

export default CartStatus;
