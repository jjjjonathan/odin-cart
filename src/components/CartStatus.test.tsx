import renderer from 'react-test-renderer';
import CartStatus from './CartStatus';

const MockAddToCart = jest.fn();
const MockRevomeFromCart = jest.fn();

describe('Cart Status Component', () => {
  it('should display success message and remove button when quantity is greater than 0', () => {
    const tree = renderer
      .create(
        <CartStatus
          quantityInCart={3}
          onAddToCart={MockAddToCart}
          onRemoveFromCart={MockRevomeFromCart}
          index={0}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should display dropdown of quantity options when quantity is 0', () => {
    const tree = renderer
      .create(
        <CartStatus
          quantityInCart={0}
          onAddToCart={MockAddToCart}
          onRemoveFromCart={MockRevomeFromCart}
          index={0}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
