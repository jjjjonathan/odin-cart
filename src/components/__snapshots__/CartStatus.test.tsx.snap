// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Status Component should display dropdown of quantity options when quantity is 0 1`] = `
<div
  className="item-form"
>
  <form
    data-index={0}
    noValidate={true}
    onSubmit={[MockFunction]}
  >
    <label
      htmlFor="quantity"
    >
      Qty:
    </label>
    <select
      id="quantity"
      name="quantity"
    >
      <option
        value="1"
      >
        1
      </option>
      <option
        value="2"
      >
        2
      </option>
      <option
        value="3"
      >
        3
      </option>
      <option
        value="4"
      >
        4
      </option>
      <option
        value="5"
      >
        5
      </option>
      <option
        value="6"
      >
        6
      </option>
    </select>
    <button
      type="submit"
    >
      Add to cart
    </button>
  </form>
</div>
`;

exports[`Cart Status Component should display success message and remove button when quantity is greater than 0 1`] = `
<div
  className="item-form"
>
  <p
    className="success"
  >
    Added 
    3
     to cart!
  </p>
  <button
    data-index={0}
    onClick={[MockFunction]}
    type="button"
  >
    Remove from cart
  </button>
</div>
`;
